test/checks/navigation/identical-links-same-purpose.js

```diff
@@ -0,0 +1,266 @@
+describe('identical-links-same-purpose tests', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var shadowSupported = axe.testUtils.shadowSupport.v1;
+	var shadowCheckSetup = axe.testUtils.shadowCheckSetup;
+	var queryFixture = axe.testUtils.queryFixture;
+	var check = checks['identical-links-same-purpose'];
+	var checkContext = axe.testUtils.MockCheckContext();
+	var options = {};
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+		checkContext.reset();
+		axe._tree = undefined;
+	});
+
+	it('returns undefined for native link with `href` but no accessible name', function() {
+		var vNode = queryFixture('<a id="target" href="/home/#/foo"></a>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when ARIA link that has no accessible name', function() {
+		var vNode = queryFixture('<span role="link" id="target"></span>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when native link has only emoji as accessible name', function() {
```

Comment:
You can probably just combine those unicode filter things into a single test. You have other tests that ensure that's sufficiently covered. That's not what this check's test should be about.